src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -212,23 +212,49 @@ namespace xrt::core::hip
   static void
   hip_memcpy_host2device_async(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)
   {
-    throw_invalid_value_if(!src, "Invalid src pointer in hipMemCpyH2DAsync!");
+    throw_invalid_value_if(!src, "src is nullptr.");
 
     auto hip_mem_info = memory_database::instance().get_hip_mem_from_addr(dst);
     auto hip_mem_dst = hip_mem_info.first;
     auto offset = hip_mem_info.second;
-    throw_invalid_value_if(!hip_mem_dst, "Invalid destination handle in hipMemCpyH2DAsync!");
-    throw_invalid_value_if(offset + size > hip_mem_dst->get_size(), "dst out of bound in hipMemCpyH2DAsync!");
+    throw_invalid_value_if(!hip_mem_dst, "Invalid destination handle.");
+    throw_invalid_value_if(offset + size > hip_mem_dst->get_size(), "dst out of bound.");
 
     auto hip_stream = get_stream(stream);
-    throw_invalid_value_if(!hip_stream, "Invalid stream handle in hipMemCpyH2DAsync!");
+    throw_invalid_value_if(!hip_stream, "Invalid stream handle.");
    
     auto s_hdl = hip_stream.get();
     auto cmd_hdl = insert_in_map(command_cache,
                                 std::make_shared<copy_buffer>(hip_stream, XCL_BO_SYNC_BO_TO_DEVICE, hip_mem_dst, src, size, offset));
     s_hdl->enqueue(command_cache.get(cmd_hdl));
   }
 
+  // fill data to dst async.
+  template<typename T> static void
+  hip_memset_async(void* dst, T value, size_t size, hipStream_t stream)
+  {
+    auto hip_mem_info = memory_database::instance().get_hip_mem_from_addr(dst);
+    auto hip_mem_dst = hip_mem_info.first;
+    auto offset = hip_mem_info.second;
+    assert(hip_mem_dst->get_type() != xrt::core::hip::memory_type::invalid);
+    throw_invalid_value_if(offset + size > hip_mem_dst->get_size(), "dst out of bound.");
+
+    auto element_size = sizeof(T);
+
+    throw_invalid_value_if((element_size != 1 && element_size != 2 && element_size != 4), "Invalid element type.");
+    throw_invalid_value_if(size % element_size != 0, "Invalid size.");
+
+    auto element_count = size / element_size;
+    std::vector<T> host_vec(element_count, value);
+
+    auto hip_stream = get_stream(stream);
+    throw_invalid_value_if(!hip_stream, "Invalid stream handle.");
+
+    auto s_hdl = hip_stream.get();
+    auto cmd_hdl = insert_in_map(command_cache,
+                                std::make_shared<copy_from_host_buffer_command<T>>(hip_stream, XCL_BO_SYNC_BO_TO_DEVICE, hip_mem_dst, host_vec, size, offset));
```

Comment:
```suggestion
    auto cmd_hdl = insert_in_map(command_cache,
                                std::make_shared<copy_from_host_buffer_command<T>>(hip_stream, XCL_BO_SYNC_BO_TO_DEVICE, hip_mem_dst, std::move(host_vec), size, offset));
```

The vector is moved into the `copy_from_host_buffer_command`.  By using move, not only do we inform compiler of intent, we also make the code explicit that `host_vec` is no longer useable in this scope.

Reply:
Done.